### ISSUE TITLE
Is it possible to combine the function of `footnote` and `inline link`?

### ISSUE THREAD

**Issue by Quantizen** [SublimeText-Markdown/MarkdownEditing#313]
What I want is that: If I have some selected text, for example, "test", and a link in the clipboard, for example, "www.test.com", then there is a `Key Binding` to make them into something like
```
[test][prefix+number] #this is in the line of "test"

[prefix+number]: www.test.com #this is at the end of document.
```
In some sense, this new function is a combination of the ones of `footnote` and `inline link`. So, it maybe not hard to add this new function.

I prefer this function to the one of `inline link`, because the former one is more friendly to read.

**Comment by felixhao28** [comment]
Usually, we use meaningful strings to name a link rather than "prefix+number". In this case, what you need can be simply achieved by `[, [, <link_name>, Alt+Shift+G`.

**Comment by Quantizen** [comment]
Okay, it's a good idea. But I have to go to the bottom and paste the link in the clipboard after the step `Alt+Shift+G` and then back. Could `Alt+Shift+G` or a similar key binding paste the link in the clipboard automatically?

**Comment by felixhao28** [comment]
Good idea. I will do something about it.

**Comment by felixhao28** [comment]
I changed how Paste As Reference works to below:

__Step 1: copy the URL__
![image](https://cloud.githubusercontent.com/assets/2417391/11834530/5fb4f8e6-a408-11e5-9010-df215e4fbc0f.png)

__Step 2: Select link word(s)__
![image](https://cloud.githubusercontent.com/assets/2417391/11834575/c1026fac-a408-11e5-8614-f4294305dbe0.png)

__Step 3: Paste As Reference__
![image](https://cloud.githubusercontent.com/assets/2417391/11834579/ce864004-a408-11e5-9661-9a85ddf1f2b8.png)

__Step4: Done__
![image](https://cloud.githubusercontent.com/assets/2417391/11834591/e5dab014-a408-11e5-9047-41767df066c3.png)

**Comment by felixhao28** [comment]
@Quantizen How does that look?

**Comment by felixhao28** [comment]
Also, since it is a feature change. @maliayas You may prefer to be notified.

**Comment by Quantizen** [comment]
Thanks for your attribution! It's awesome! I cannot wait to have a try.

**Comment by Quantizen** [comment]
And as I test in Windows, `Ctrl+Win+R` will directly generate 
```
[test]: https://github.com/SublimeText-Markdown/MarkdownEditing/issues/313
```

**Comment by felixhao28** [comment]
> then I type test and hit Enter again, but nothing happens. Where am I wrong?

That feature is broken in v2.1.4 (current version). It is already fixed in v2.1.5. The fix and the new behavior of `paste as reference` will be shipped along with v2.1.5. This may take weeks to happen. So if you need this feature now, you can follow the manual installation guide in readme (assuming you know git).

**Comment by Quantizen** [comment]
Thanks! I download v2.1.5, it works fine!
